modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```
@@ -338,93 +338,98 @@ private void updateSlotAssignmentMap(String queueName, Slot allocatedSlot, Strin
-                        nodeToLastPublishedIDMap.set(nodeId, lastMessageIdInTheSlot);
-                    }
 
 
+                    }
 
-                }
+                } else {
+                    /**
+                     * Update the slotIDMap only if the last assigned message ID is less than the new start message ID
+                     */
+                    messageIdSet.add(lastMessageIdInTheSlot);
+                    wrapper.setLongTreeSet(messageIdSet);
+                    slotIDMap.set(queueName, wrapper);
+                    if (log.isDebugEnabled()) {
+                        log.debug("No overlapping slots found during slot submit "+ startMessageIdInTheSlot + " to : " +
+                                lastMessageIdInTheSlot+". Added msgID " +
+                                lastMessageIdInTheSlot + " to slotIDMap");
+                    }
 
-            } else {
-                /**
-                 * Update the slotIDMap only if the last assigned message ID is less than the new start message ID
-                 */
-                messageIdSet.add(lastMessageIdInTheSlot);
-                wrapper.setLongTreeSet(messageIdSet);
-                slotIDMap.set(queueName, wrapper);
-                if (log.isDebugEnabled()) {
-                    log.debug("No overlapping slots found during slot submit "+ startMessageIdInTheSlot + " to : " +
-                            lastMessageIdInTheSlot+". Added msgID " +
-                            lastMessageIdInTheSlot + " to slotIDMap");
+                    //record last published message ID
+                    nodeToLastPublishedIDMap.set(nodeId, lastMessageIdInTheSlot);
                 }
-
-                //record last published message ID
-                nodeToLastPublishedIDMap.set(nodeId, lastMessageIdInTheSlot);
             }
+        } else {
+            log.warn("The server is shutting down and therefore the cluster is inactive. Therefore updating the slot " +
```

Here you have to check whether the server is shutting down I guess. Its better if you can catch the HazelcastInstanceNotActiveException and if the server is shutting down ignore the exception. 

} catch (HazelcastInstanceNotActiveException e) {
                           if(!serverlsShuttingDown){
                                 log.error("Hazelcast instance is not active and therefore cannot update the slot manager", e);
                           }
                        // Ignore this exception if the server is shutting down
                    }